examples/user_guide/Reactive_Expressions.ipynb

```diff
@@ -0,0 +1,258 @@
+   "id": "243afd62-2f10-4dd3-a264-4d9a895c3088",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "p = Parameters()\n",
+    "\n",
+    "p.param.a.reactive() + p.param.b.reactive()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f62fe083-50c9-4739-aaec-81fb31f98241",
+   "metadata": {},
+   "source": [
+    "The resulting reactive expression now reflects the result of this operation and will update automatically when one of the inputs to the operation changes, e.g. if we update parameter `a`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f71b9341-31bb-437f-90ab-3fb5da607f0a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "p.a = 5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "824dc59e-aea0-404e-b3c3-be134dffa8e0",
+   "metadata": {},
+   "source": [
+    "### Using objects"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "26ad333e-0d05-4591-8492-24b641171f4d",
+   "metadata": {},
+   "source": [
+    "Note that the input to a reactive expression can be anything, e.g. we can create a `reactive` object from a static value, such as a string:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9b061295-a28c-4e13-9b8c-54f9070a0a4a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "string_template = param.reactive('Hello {name}!')\n",
+    "\n",
+    "string_template"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "05af1fa3-bcea-4c55-809f-beb8e4049e90",
+   "metadata": {},
+   "source": [
+    "The `reactive` object now acts like a string so we can use the `.format` method to fill in the `string_template`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "56d8047b-b737-4b03-86a1-24b5b4d49bd3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "name = param.reactive('world')\n",
+    "\n",
+    "str_expr = string_template.format(name=name)\n",
+    "\n",
+    "str_expr"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3218ab95-0261-4cd9-a4fe-6f552baafdb3",
+   "metadata": {},
+   "source": [
+    "To update the input to a `reactive` object we can use the `.set` method:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d4842f2f-8d17-41eb-96f0-668d0ee545d1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "name.set('there')\n",
+    "\n",
+    "str_expr"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "227a230b-98b2-4097-8a6d-798ddb63b74a",
+   "metadata": {},
+   "source": [
+    "## Limitations\n",
+    "\n",
+    "A `reactive` proxy behaves like the underlying object it is wrapping, but only to the extent that Python allows. Certain operations cannot be implemented, e.g. Python will now allow the `len` operation to return anything but a integer. Therefore `reactive` implements certain operations as methods, these include:\n",
+    "\n",
+    "- `.bool()`: Tests the truthiness of the object\n",
```

Comment:
Don't you risk having to implement an ever growing list of methods here?

Have you considered having a module of special reactive functions instead of adding these as methods?

```
param.reactive.special_functions(some_reactive_object)
```

The pro would be that you keep the `Reactive` api simpler and you can explain where all the "special" functions can be discovered. The con would be the lenghty expression or that users would have to import the module.

Reply:
Yeah, this is a good point. The idea behind these special functions is that I'm trying to cover the full breadth of Python built-in operators and statements, i.e. ideally the entire namespace of the `operator` module in Python should be supported in some form. So from that perspective the set of possible operators is well defined and not "ever growing".